codec/src/test/java/io/netty/handler/codec/compression/JdkZlibTest.java

```diff
@@ -35,4 +47,45 @@ protected ZlibDecoder createDecoder(ZlibWrapper wrapper) {
     public void testZLIB_OR_NONE3() throws Exception {
         super.testZLIB_OR_NONE3();
     }
+
+    @Test
+    // verifies backward compatibility
+    public void testConcatenatedStreamsReadFirstOnly() throws IOException {
+        EmbeddedChannel chDecoderGZip = new EmbeddedChannel(createDecoder(ZlibWrapper.GZIP));
+
+        try {
+            byte[] bytes = IOUtils.toByteArray(getClass().getResourceAsStream("/multiple.gz"));
+
+            chDecoderGZip.writeInbound(Unpooled.copiedBuffer(bytes));
+            Queue<Object> messages = chDecoderGZip.inboundMessages();
+            assertEquals(1, messages.size());
+
+            ByteBuf msg = (ByteBuf) messages.poll();
+            assertEquals("a", msg.toString(Charset.forName("UTF-8")));
+            ReferenceCountUtil.release(msg);
+        } finally {
+            chDecoderGZip.close();
+        }
+    }
+
+    @Test
+    public void testConcatenatedStreamsReadFully() throws IOException {
+        EmbeddedChannel chDecoderGZip = new EmbeddedChannel(new JdkZlibDecoder(true));
+
+        try {
+            byte[] bytes = IOUtils.toByteArray(getClass().getResourceAsStream("/multiple.gz"));
+
+            chDecoderGZip.writeInbound(Unpooled.copiedBuffer(bytes));
+            Queue<Object> messages = chDecoderGZip.inboundMessages();
+            assertEquals(2, messages.size());
+
+            for (String s : Arrays.asList("a", "b")) {
+                ByteBuf msg = (ByteBuf) messages.poll();
+                assertEquals(s, msg.toString(Charset.forName("UTF-8")));
```

Comment:
use `CharsetUtil.UTF_8`

Reply:
Done. Due to JDK 6 compilation level, `StandardCharsets.UTF-8` can't be used.